### PR TITLE
Add i18n-tasks as a dev dependency

### LINKED ISSUE #876: i18n refactor and testing
NOTE: This issue is suitable for someone new to the community.

### Descriptive summary

Refactor to remove unused translations and add missing translations.

### Rationale

gem i18n-tasks provides a command line that checks for...
* missing translations (those in code but not in locale file) using `i18n-tasks missing`
* unused translations (those in locale file but not in code) using `i18n-tasks unused`

These can be used to identify potential translations to add or remove.  Unused translations can be added as exceptions to [config/i18n-tasks.yml](https://github.com/projecthydra-labs/hyrax/blob/i18n_testing/config/i18n-tasks.yml) if they should be retained even though they are unused.

The gem also provides a spec test that validates that the missing translations and unused translations lists are empty.  This can be used to confirm that new issues with i18n are not introduced in the future.

### Expected Behavior

`i18n-tasks missing` returns empty array indicating there are no missing translations
`i18n-tasks unused` returns empty array indicating there are no unused translations

### Actual Behavior

`i18n-tasks missing` returns 31
`i18n-tasks unused` returns 152

NOTE: Some of the missing may actually be coming from other gems.  Additional investigation required.  Also, some of the unused need investigation before removing.

### Additional Information

Initial setup of i18n-tasks in branch [i18n-testing](https://github.com/projecthydra-labs/hyrax/tree/i18n_testing)

[Documentation for i18n-tasks gem](https://github.com/glebm/i18n-tasks)

### PR DESCRIPTION
**Pull request by mjgiarlo**
Now that Hyrax supports seven languages, i18n-tasks can help us improve translations over time and keep them in sync.

Fixes #876

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: hyrax.gemspec [hunk 1]
```diff
@@ -91,6 +91,7 @@ EOF
   spec.add_development_dependency 'shoulda-matchers', '~> 3.1'
   spec.add_development_dependency 'rails-controller-testing', '~> 0'
   spec.add_development_dependency 'webmock'
+  spec.add_development_dependency 'i18n-tasks'
```
Comment (jcoyne) [inline hyrax.gemspec]: Do we need to require this in the app if it's a dependency?  If not, is it truly a dependency?

**Comment by mjgiarlo** [comment]
@jcoyne re-review?
